src/main/java/org/mule/extension/runtime/Expirable.java

```diff
@@ -0,0 +1,20 @@
+package org.mule.extension.runtime;
+
+/**
+ * An entity which can or cannot be considered elegible for expiration
+ * according to a {@link ExpirationPolicy}
+ *
+ * @since 1.0
+ */
+public interface Expirable
+{
+
+    /**
+     * Determins if {@code this} instance should be expired based on the given
+     * {@code expirationPolicy}
+     *
+     * @param expirationPolicy a {@link ExpirationPolicy}
+     * @return {@code true} if {@code this} can be expired. {@code false} otherwise
```

Comment:
To me, something "expired" is something that was declared to be "unusable". If the object was indeed disposed as a result of detecting that was expired is a separated thing.
So, isExpired is correct to me and the javadoc needs that little change